pkg/kubefed2/federate/federate.go

```diff
@@ -166,12 +160,16 @@ type fedResources struct {
 	FedResource *unstructured.Unstructured
 }
 
-func GetFedResources(hostConfig *rest.Config, qualifiedTypeName, qualifiedName ctlutil.QualifiedName) (*fedResources, error) {
+func GetFedResources(hostConfig *rest.Config, qualifiedTypeName, qualifiedName ctlutil.QualifiedName, outputYAML bool) (*fedResources, error) {
 	typeConfig, err := lookupTypeDetails(hostConfig, qualifiedTypeName)
 	if err != nil {
 		return nil, err
 	}
 
+	if typeConfig.Spec.Target.Kind == ctlutil.NamespaceKind && !outputYAML {
```

Comment:
Would it make sense to move this to `CreateFedResource`?

Reply:
Yes, it does. Updated!